src/main/scala/com/softwaremill/realworld/articles/ArticlesService.scala

```diff
@@ -56,7 +49,8 @@ class ArticlesService(articlesRepository: ArticlesRepository, usersRepository: U
         .fail(Unauthorized(s"You're not an author of article that you're trying to update"))
         .when(user.username != oldArticle.author.username)
       updatedArticle = updateArticleData(oldArticle, articleUpdateData)
-      _ <- articlesRepository.updateBySlug(updatedArticle, oldArticle.slug)
+      articleId <- articlesRepository.findArticleIdBySlug(slug)
```

Comment:
I think such operations are typically run in a single transation - and here we should do the same

Reply:
I agree we should take care about transactions in this and other endpoints in the project. As we agreed - there will be the issue to cover that later.